src/Refactoring-UI/RePushUpMethodDriver.class.st

```diff
@@ -0,0 +1,241 @@
+		'superclass',
+		'notOverrides',
+		'notInstVarRefs',
+		'notSharedVarRefs',
+		'notSupersendsReceived',
+		'notSupersendsSent'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'initialization' }
+RePushUpMethodDriver class >> model: model scopes: refactoringScopes pushUpMethods: methods [
+
+	^ self new
+		  model: model
+		  scopes: refactoringScopes
+		  pushUpMethods: methods
+]
+
+{ #category : 'ui - dialogs' }
+RePushUpMethodDriver >> alertFailedPreconditions [
+	SpAlertDialog new
+		title: 'Push Up failed: Required preconditions were not met';
+		label:  self failedPreconditionsErrorString;
+		acceptLabel: 'Accept';
+		openDialog
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> breakingChoices [
+
+	^ refactoring failedBreakingChangePreconditions collect: [ :each |
+		each class strategyChoiceClass new driver: self ]
+]
+
+{ #category : 'actions' }
+RePushUpMethodDriver >> browseOverrides [
+
+	| overrides |
+	overrides := notOverrides violators.
+	methods do: [:method | 
+		StMessageBrowser
+		browse: (overrides collect: [ :ref | ref realClass methodNamed: method selector ])
+		asImplementorsOf: method selector ]
+	
+]
+
+{ #category : 'actions' }
+RePushUpMethodDriver >> browseSuperSendsSent [
+	| overrides |
+	overrides := notSupersendsSent violators.
+	
+	methods do: [:method | 
+		StMessageBrowser
+		browse: (overrides collect: [ :ref | ref compiledMethod ])
+		asImplementorsOf: method selector ]
+	
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> changes [
+	"it should NOT be generate changes."
+	^  refactoring privateTransform
+
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> configureAndRunRefactoring [
+
+	self configureRefactoring.
+
+	refactoring failedApplicabilityPreconditions ifNotEmpty: [
+		^ self alertFailedPreconditions ].
+	
+	self setBreakingChangesPreconditions.
+	refactoring failedBreakingChangePreconditions
+		ifEmpty: [ self applyChanges ]
+		ifNotEmpty: [ self handleBreakingChanges ]
+]
+
+{ #category : 'resources' }
+RePushUpMethodDriver >> configureRefactoring [
+
+	refactoring := RBPullUpMethodRefactoring
+		               model: model
+		               pullUp: (methods collect: [ :each | each selector ])
+		               from: class name
+		               to: superclass name
+]
+
+{ #category : 'ui - dialogs' }
+RePushUpMethodDriver >> failedPreconditionsErrorString [
+
+	^ String streamContents: [ :stream |
+		  refactoring failedApplicabilityPreconditions do: [ :cond |
+			  cond check ifFalse: [
+				  cond violationMessageOn: stream.
+				  stream cr ] ] ]
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> gatherUserInput [
+	self selectMethodsAndClass
+]
+
+{ #category : 'execution' }
+RePushUpMethodDriver >> handleBreakingChanges [
+
+	| presenter changes |
+	
+	"Display previous transformations that will fix the fixable breaking conditions"
+	changes := self breakingChoices.
```

Comment:
The variable name should be `choices` right?

Reply:
Oh yes! easy to fix mistake :)